From f5b972f525925ef408a43bf393f983eccaf6c802 Mon Sep 17 00:00:00 2001
From: Fabio Fantoni <fantonifabio@tiscali.it>
Date: Mon, 9 Jan 2023 20:03:50 +0100
Subject: [PATCH v2] documentation: fix Generic Block Device Capability

When adding documentation for blkfilter, new lines of documentation was
added in the file include/linux/blkdev.h. Sergei Shtepa spotted that
this changed Documentation/block/capability.rst output.
Checking documentation of previous versions, for example
https://www.kernel.org/doc/html/v5.10/block/capability.html I also saw
that the output of recent versions was already "broken".
This patch restore the output only relate to generic block device
capability and also added the hexadecimal values that was removed from
doc comment in commit 430cc5d (block: cleanup the GENHD_FL_*
definitions).
This will make this documentation page return to "documents the sysfs
file block/<disk>/capability" including the printed in hexadecimal
value.

Co-developed-by: Sergei Shtepa <sergei.shtepa@veeam.com>
Signed-off-by: Sergei Shtepa <sergei.shtepa@veeam.com>
Signed-off-by: Fabio Fantoni <fantonifabio@tiscali.it>
---
Note: v1 was included in blksnap v2 serie

Changes from v1:
- Documentation/block/capability.rst: removed other blkdev.h functions
  and keep only ":DOC: genhd capability flags"
- added changes of include/linux/blkdev.h
---
 Documentation/block/capability.rst |  1 +
 include/linux/blkdev.h             | 10 +++++-----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/Documentation/block/capability.rst b/Documentation/block/capability.rst
index 2ae7f064736a..e5d58e0d81ba 100644
--- a/Documentation/block/capability.rst
+++ b/Documentation/block/capability.rst
@@ -8,3 +8,4 @@ This file documents the sysfs file ``block/<disk>/capability``.
 capabilities a specific block device supports:
 
 .. kernel-doc:: include/linux/blkdev.h
+    :DOC: genhd capability flags
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 43d4e073b111..71ca547ab19f 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -74,16 +74,16 @@ struct partition_meta_info {
 /**
  * DOC: genhd capability flags
  *
- * ``GENHD_FL_REMOVABLE``: indicates that the block device gives access to
+ * ``GENHD_FL_REMOVABLE`` (0x01): indicates that the block device gives access to
  * removable media.  When set, the device remains present even when media is not
  * inserted.  Shall not be set for devices which are removed entirely when the
  * media is removed.
  *
- * ``GENHD_FL_HIDDEN``: the block device is hidden; it doesn't produce events,
- * doesn't appear in sysfs, and can't be opened from userspace or using
- * blkdev_get*. Used for the underlying components of multipath devices.
+ * ``GENHD_FL_HIDDEN`` (0x02): the block device is hidden; it doesn't produce
+ * events, doesn't appear in sysfs, and can't be opened from userspace or
+ * using blkdev_get*. Used for the underlying components of multipath devices.
  *
- * ``GENHD_FL_NO_PART``: partition support is disabled.  The kernel will not
+ * ``GENHD_FL_NO_PART`` (0x04): partition support is disabled. The kernel will not
  * scan for partitions from add_disk, and users can't add partitions manually.
  *
  */
-- 
2.25.1

